python/lsst/ts/standardscripts/maintel/m1m3/check_actuators.py

```diff
@@ -0,0 +1,226 @@
+
+        # Getting actuators ids
+        self.actuators = self.m1m3_actuator_ids
+
+    async def assert_feasibility(self):
+        """Verify that the system is in a feasible state before
+        running bump test.
+        """
+
+        for comp in self.mtcs.components_attr:
+            if comp != "mtm1m3":
+                self.log.debug(f"Ignoring component {comp}.")
+                setattr(self.mtcs.check, comp, False)
+
+        await asyncio.gather(
+            self.mtcs.assert_all_enabled(),
+            self.mtcs.assert_liveliness(),
+        )
+
+    @classmethod
+    def get_schema(self, cls):
+
+        m1m3_actuator_ids_str = ', '.join(str(fa) for fa in self.m1m3_actuator_ids)
+
+        url = "https://github.com/lsst-ts/"
+        path = (
+            "ts_externalscripts/blob/main/python/lsst/ts/standardscripts/"
+            "maintel/m1m3/check_actuators.py"
+        )
+        schema_yaml = f"""
+        $schema: http://json-schema.org/draft-07/schema#
+        $id: {url}{path}
+        title: CheckAcutators v1
+        description: Configuration for Maintel bump test SAL Script.
+        type: object
+        properties:
+            actuators:
+                description: Actuators to run the bump test.
+                oneOf:
+                  - type: array
+                    items:
+                      type: number
+                      enum: [{m1m3_actuator_ids_str}]
+                    minItems: 1
+                    uniqueItems: true
+                    additionalItems: false
+                  - type: string
+                    enum: ["all"]
+                default: "all"
+        additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+   
+    async def configure(self, config):
+        """Configure the script.
+        
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Configuration
+        """
+
+        # Actuators defined as "all" (ID) or a list of actuators (INDEX)
+        self.actuators = (
+            self.m1m3_actuator_ids if config.actuators == "all" else config.actuators
+        )
+
+        # Check if actuators are valid
+        if not all([actuator in self.m1m3_actuator_table_id for actuator in self.actuators]):
+            raise ValueError("Invalid actuator ID.")
+
+    def set_metadata(self, metadata):
+        
+        metadata.duration = self.timeout_one_bump * (self.total_actuators + self.total_sec_actuators) 
+
+    def is_actuator_dual(self, actuator_id: int) -> bool:
+        
+        # Converting actuator id to index
+        actuator_index = self.mtcs.get_m1m3_actuator_index(actuator_id)
+
+        is_dual = FATABLE[actuator_index][M1M3FATable.FATABLE_TYPE] == 'DAA'
+
+        return is_dual
+
+    async def run(self):
+
+        await self.assert_feasibility()
+        start_time = time.time()
+
+        # Get M1M3 detailed state
```

Comment:
I think you can skip logging the detailed state.

Reply:
The original idea of this script according to its ticket ([SITCOM-745](https://jira.lsstcorp.org/browse/SITCOM-745)) was:

_The SAL Script will consider only the Bump Test itself. It should check if M1M3 is in ACTIVEENGINEERING or in PARKEDENGINEERING detailed state, but it should NOT change the detailed state. This way, we can have a single script to run both test cases and avoid duplication._

That is why I'm logging the M1M3 detailed state here...

Reply:
If you do the check of the detailed state in `asset_feasibility` it will raise an exception when the detailed state is not the desired one, and continue if it is ok. I think this is enough information.

If you insist in logging it, please add a timeout in `aget` to avoid blocking here for ever if there's ever loss of historical data.